Google.Api.Gax.Grpc/CallSettingsExtensions.cs

```diff
@@ -61,28 +61,49 @@ public static CallSettings MergedWith(this CallSettings original, CallSettings o
             settings == null
                 ? CallSettings.FromCallCredentials(credentials)
                 : new CallSettings(settings.CancellationToken, credentials,
-                    settings.Timing, settings.HeaderMutation,
+                    settings.Expiration, settings.Retry, settings.HeaderMutation,
                     settings.WriteOptions, settings.PropagationToken,
                     settings.ResponseMetadataHandler, settings.TrailingMetadataHandler);
 
         /// <summary>
-        /// Returns a new <see cref="CallSettings"/> with the specified call timing,
+        /// Returns a new <see cref="CallSettings"/> with the specified expiration,
         /// merged with the (optional) original settings specified by <paramref name="settings"/>.
         /// </summary>
         /// <param name="settings">Original settings. May be null, in which case the returned settings
         /// will only contain call timing.</param>
-        /// <param name="timing">Call timing for the new call settings.
+        /// <param name="expiration">Expiration to use in the returned settings, possibly as part of a retry. Must not be null.</param>
+        /// <returns>A new set of call settings, or null if both parameters are null.</returns>
+        public static CallSettings WithExpiration(
+            this CallSettings settings,
+            Expiration expiration)
+        {
+            GaxPreconditions.CheckNotNull(expiration, nameof(expiration));
+            return settings == null
+                ? CallSettings.FromExpiration(expiration)
+                : new CallSettings(settings.CancellationToken, settings.Credentials,
+                    expiration, settings.Retry, settings.HeaderMutation,
+                    settings.WriteOptions, settings.PropagationToken,
+                    settings.ResponseMetadataHandler, settings.TrailingMetadataHandler);
+        }
+
+        /// <summary>
+        /// Returns a new <see cref="CallSettings"/> with the specified retry settings,
+        /// merged with the (optional) original settings specified by <paramref name="settings"/>.
+        /// </summary>
+        /// <param name="settings">Original settings. May be null, in which case the returned settings
+        /// will only contain call timing.</param>
+        /// <param name="retry">Call timing for the new call settings.
         /// This may be null, in which case any call timing in <paramref name="settings"/> are
         /// not present in the new call settings. If both this and <paramref name="settings"/> are null,
         /// the return value is null.</param>
         /// <returns>A new set of call settings, or null if both parameters are null.</returns>
-        public static CallSettings WithCallTiming(
+        public static CallSettings WithRetry(
```

Comment:
Handling of a null parameter, `retry` in this case, is inconsistent with the `WithExpiration` method above.
I'm not sure which is best, but they should probably be the same!